dask/benchmarks/order.py

```diff
@@ -0,0 +1,185 @@
+
+    def time_order_svd(self):
+        order(self.dsk_svd)
+
+
+class OrderRechunkTranspose(DaskSuite):
+    def setup(self):
+        a = da.random.normal(size=(4e6, 30e2), chunks=(2e4, 3e1))
+        a = a.rechunk((int(1e4 / 10), int(30e2)))
+        b = a.T.dot(a)
+        self.dsk_rechunk_transpose = collections_to_dsk(b)
+
+    def time_order04(self):
+        order(self.dsk_rechunk_transpose)
+
+
+class OrderLinalgSolves(DaskSuite):
+    def setup(self):
+        n = 1000
+        x = da.random.normal(size=(n, 100), chunks=(1, 100))
+        y = da.random.normal(size=(n,), chunks=(1,))
+        xy = (x * y[:, None]).cumsum(axis=0)
+        xx = (x[:, None, :] * x[:, :, None]).cumsum(axis=0)
+        beta = da.stack(
+            [da.linalg.solve(xx[i], xy[i]) for i in range(xx.shape[0])], axis=0
+        )
+        ey = (x * beta).sum(axis=1)
+        self.dsk_linalg = collections_to_dsk(ey)
+
+    def time_order_linalg_solve(self):
+        order(self.dsk_linalg)
+
+
+class OrderFullLayers(DaskSuite):
+    def setup(self):
+        self.dsk1 = fully_connected_layers(1, 50000)
+        self.dsk2 = fully_connected_layers(2, 10000)
+        self.dsk3 = fully_connected_layers(10, 1000)
+        self.dsk4 = fully_connected_layers(100, 20)
+        self.dsk5 = fully_connected_layers(500, 2)
+        self.dsk6 = fully_connected_layers(9999, 1)
+        self.dsk7 = fully_connected_layers(50000, 1)
+
+    def time_order_full_1_50000(self):
+        order(self.dsk1)
+
+    def time_order_full_2_10000(self):
+        order(self.dsk2)
+
+    def time_order_full_10_1000(self):
+        order(self.dsk3)
+
+    def time_order_full_100_20(self):
+        order(self.dsk4)
+
+    def time_order_full_500_2(self):
+        order(self.dsk5)
+
+    def time_order_full_9999_1(self):
+        order(self.dsk6)
+
+    def time_order_full_50000_1(self):
+        order(self.dsk7)
+
+
+class OrderLinearWithDanglers(DaskSuite):
+    def setup(self):
+        self.dsk1 = {(0, 0): (f,)}
+        for i in range(1, 10000):
+            for j in range(2):
+                self.dsk1[(i, j)] = (f, (i - 1, 0))
+
+        self.dsk2 = {(0, 0): (f,)}
+        for i in range(1, 5000):
+            for j in range(5):
+                self.dsk2[(i, j)] = (f, (i - 1, 0))
+
+    def time_order_linear_danglers_2(self):
+        order(self.dsk1)
+
+    def time_order_linear_danglers_5(self):
+        order(self.dsk2)
+
+
+class OrderLinearFull(DaskSuite):
+    def setup(self):
+        # Although not a realistic DAG, this is cleverly constructed
+        # to stress a current weakness of `order`.
+        self.dsk = {0: (f,)}
+        prev = (f, 0)
+        for i in range(1, 1000):
+            self.dsk[i] = prev
+            prev += (i,)
+
+    def time_order_linear_full(self):
+        order(self.dsk)
+
+
+class OrderManySubgraphs(DaskSuite):
+    """This tests behavior when there are few or many disconnected subgraphs"""
+
+    def setup(self):
```

Comment:
May be able to use a parametrized benchmark here. https://asv.readthedocs.io/en/stable/writing_benchmarks.html#parameterized-benchmarks

I think the only real benefit would be shorter setup times, as only the one `create_disconnected_subgraph` would be called. Possibly nicer reporting.

Reply:
Done (and applied elsewhere).